Products/ZenModel/Device.py

```diff
@@ -1909,11 +1910,32 @@ def renameDevice(self, newId=None, REQUEST=None):
             parent.manage_renameObject(oldId, newId)
             self.restoreCurrentProdStates(currProdStates)
             self.setLastChange()
+
+            # Replace the old id in performance data with the new id.
+            # See ZEN-27329.
+            self.amendPerfDataAfterRename(oldId, newId)
```

Comment:
Should this be optional? That is should a user be able to rename a device with out "moving" metric data?